src/corelibs/U2Core/src/gobjects/MultipleAlignmentObject.cpp

```diff
@@ -737,6 +737,21 @@ void MultipleAlignmentObject::releaseState() {
     }
 }
 
+bool MultipleAlignmentObject::hasNonTrailingGap() const {
+    const QList<QVector<U2MsaGap>> &listGapModel = getGapModel();
+    for (int i = 0; i < getRowCount(); i++) {
+        if (listGapModel[i].isEmpty()) {
```

Comment:
```!istGapModel[i].isEmpty()``` alone is enough to check if the row has non-trailing gaps.

Reply:
Here you check that the row has no 'inner' gaps.
There are 3 types of gaps: leading, inner, trailing.
What you need to check here is that there are either leading or inner gaps. In this case you can remove gaps from the row.

To check if you can remove any gaps you do not need any method at all: use ```!getGaps().isEmpty()```